### ISSUE TITLE
Searching for a key

### ISSUE THREAD

**Issue by ashikvarma11** [ashikvarma11/tree-traverse#13]
Use any traversal method already implemented.

- Trace the path of searching and once the value is detected, highlight that node.
- If not found, display proper message.

**Comment by sambhavjain9138** [comment]
Heyy @ashikvarma11 
I would like to contribute to the issue. Can you please assign it to me?

Regards,
Sambhav

**Comment by ashikvarma11** [comment]
@sambhavjain9138  Sure. go ahead. Ping me in case of any doubts.

**Comment by sambhavjain9138** [comment]
Heyy @ashikvarma11 
Should i implement the function in the tree.component.js file or should i add the functionality to tree class as a new function?

Regards,
Sambhav

**Comment by ashikvarma11** [comment]
@sambhavjain9138  Whichever is easy to implement. As this involves traversal as well, I believe, it is easier to implement it in tree class. The traversal should stop when you find the key. You could call a specific traversal function (inorder,preorder or postorder )each time user clicks on the search button.

**Comment by sambhavjain9138** [comment]
Thank for the guidance. I was thinking to fetch the list of the node in the tree in inorder traversal and then by finding the node by binary search. will that be fine?

Also in case the node is not found, where should i display the error msg?

Regards,
Sambhav

**Comment by ashikvarma11** [comment]
@sambhavjain9138  If you fetch the node list and then do binary search, how will we show the trace path? This task is a simple one. What you need t do is call a traversal method and let it run until it finds the given value/key. At this point, instead of continuing the recursive traversal, you need to exit out of the function. Then, the animation will stop. Also need to highlight that node with a different color, if found.  And for showing a message, you could add a popup with a good design. Better to use some libraries. I prefer some light-weight ones. If you can create your own function to show/hide popup, better. The popup shouldnot be a large one. Instead , show it either on the left/right side as a box. Like the Windows notification

**Comment by sambhavjain9138** [comment]
@ashikvarma11 Okay got it. I forgot that it is a binary search tree. I hope you wont mind me implementing a new search algorithm without using existing traversal algorithms based on the value of current node? I can implement it in the tree class.

Regards,
Sambhav

**Comment by ashikvarma11** [comment]
@sambhavjain9138  Go ahead.

**Comment by sambhavjain9138** [comment]
@ashikvarma11 I implemented the algorithm. But adding a search box and a button is taking space, minimizing and disturbing the other header content. 
I consider a solution for the same can be made by using same input text box for all the 3 tasks. should i go for it?

![snap](https://user-images.githubusercontent.com/43277337/95000957-0ab0b500-05e3-11eb-9673-d79adf5b9f0c.PNG)

making it look somewhat like this. 

![snap](https://user-images.githubusercontent.com/43277337/95001026-bf4ad680-05e3-11eb-8cc3-bab1aa061fa1.PNG)

**Comment by ashikvarma11** [comment]
@sambhavjain9138  Yes. that would be good. Create a new issue for this. Also, make the buttons responsive. Right now, `flex:wrap` is not present in button styles. The whole row is in `flex`. Maybe, we should use `display:grid` for just the buttons, so that while resizing the window, buttons would align to write and wraps there itself.

**Comment by sambhavjain9138** [comment]
@ashikvarma11 Can i make the changes other than that of CSS because i have very less idea of frontend part?  So incase it is not a problem, i can implement a search algorithm and also make same text option for the three functionalities i.e. find, insert and remove.

**Comment by ashikvarma11** [comment]
@sambhavjain9138 Yes. That is fine. But you should create a separate issue for this. The updated Ui will be merged soon. Take an update.

**Comment by sambhavjain9138** [comment]
Okay Sure I'll do so. Thanks!

**Comment by sambhavjain9138** [comment]
Let me know if the changes needs any changes. Coz i am not sure of the reason of the conflict in the pull request.

**Comment by sambhavjain9138** [comment]
I actually disturbed the symmetrical arrangement of the navbar. In case it is convenient with you, i can push the input box to the first place and search button to the second place.
